server/odc-service/src/main/java/com/oceanbase/odc/service/collaboration/environment/EnvironmentService.java

```diff
@@ -155,6 +156,18 @@ public Environment detail(@NonNull Long id) {
         return innerDetail(id);
     }
 
+    @PreAuthenticate(actions = "select", resourceType = "ODC_ENVIRONMENT", indexOfIdParam = 0)
```

Comment:
what does this mean?

Reply:
perform some authentications of select, refer to the create method above

Reply:
You don't need to add this annotation when you use this method internally. Add `@SkipAuthorize("odc internal usage")` instead.
BTW. the action is not "select", it should be read. Refer to other methods before you write.

Reply:
> You don't need to add this annotation when you use this method internally. Add `@SkipAuthorize("odc internal usage")` instead. BTW. the action is not "select", it should be read. Refer to other methods before you write.

OK, I only refer to the `create` method here.